File: solutions/php/0189-rotate-array.php
<?php

class Solution
{
    /**
     * @param int[] $nums
     * @param int $k
     * @return NULL
     */
    public function rotate(array &$nums, int $k)
    {
        $len = count($nums);

        $k = $k % $len;

        if ($k === 0) {
            return;
        }

        $nums = array_merge(array_slice($nums, $len - $k, $k), array_slice($nums, 0, $len - $k));
    }
}

File: solutions/go/0189-rotate-array.go
package main

// Another solution using append()
func rotate2(nums []int, k int) {
	division := len(nums) - k%len(nums)
	copy(nums, append(nums[division:], nums[:division]...))
}

func rotate(nums []int, k int) {
	k %= len(nums)

	if k == 0 {
		return
	}

	count := len(nums)
	temp := make([]int, count)
	copy(temp, nums)

	for i := 0; i < count; i++ {
		j := (count - k + i) % count
		nums[i] = temp[j]
	}
}
